pkgs/development/python-modules/txtai/default.nix

```
@@ -0,0 +1,145 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, pythonRelaxDepsHook
+# propagated build input
+, faiss
+, torch
+, transformers
+, huggingface-hub
+, numpy
+, pyyaml
+, regex
+# optional-dependencies
+, aiohttp
+, fastapi
+, uvicorn
+# , apache-libcloud
+, rich
+, duckdb
+, pillow
+, networkx
+, python-louvain
+, onnx
+, onnxruntime
+, soundfile
+, scipy
+, ttstokenizer
+, beautifulsoup4
+, nltk
+, pandas
+, tika
+, imagehash
+, timm
+, fasttext
+, sentencepiece
+, accelerate
+, onnxmltools
+, annoy
+, hnswlib
+, scikit-learn
+, sentence-transformers
+, croniter
+, openpyxl
+, requests
+, xmltodict
+# native check inputs
+, unittestCheckHook
+}:
+let
+  version = "5.5.1";
+  api = [ aiohttp fastapi uvicorn ];
+  # cloud = [ apache-libcloud ];
```

```suggestion
```

---

I like the commented out packages so that someone packagin something knows that it's needed somewhere.
It also gives improvement hints for people. Otherwise people are left with the task of parsing the original setup.py or project.toml all anew.

---

In that case I usually do the following

```suggestion
  cloud = [
    # TODO: apache-libcloud
  ];
```

---

> if someone tries to use it, they will run into an unexpected error.

I toyed with using `throw` or `abort`, but I couldn't get it to work. I prefer hexa's suggestion, but it's not a huge issue to me if you think that error is too big of a problem.